patches/server/0806-prevent-unintended-light-block-manipulation.patch

```diff
@@ -0,0 +1,25 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Mon, 13 Sep 2021 18:55:45 -0700
+Subject: [PATCH] prevent unintended light block manipulation
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/LightBlock.java b/src/main/java/net/minecraft/world/level/block/LightBlock.java
+index 7f4acaab7e77f5184c8f8a30ed13b34949780227..251973703ae77de7ca7b55128952c8ceb663ca16 100644
+--- a/src/main/java/net/minecraft/world/level/block/LightBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/LightBlock.java
+@@ -46,8 +46,14 @@ public class LightBlock extends Block implements SimpleWaterloggedBlock {
+     @Override
+     public InteractionResult use(BlockState state, Level world, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit) {
+         if (!world.isClientSide) {
++            if (player.getItemInHand(hand).getItem() == Items.LIGHT && (player.getServer().getPlayerList().isOp(player.gameProfile) || ((net.minecraft.server.level.ServerPlayer) player).gameMode.getGameModeForPlayer().isCreative())) { // Paper
```

Comment:
Just having the item in hand should be good enough

Reply:
I disagree, that means if your in adventure mode you can still modify the block.

Reply:
i see, adding a check for adventure should be good then

it should be the case that if a player can place/destroy a light block (it can be destroyed by being placed over by another block) then editing it should not require additional permissions

Reply:
Ok, made the check if item is LIGHT and player's gamemode is not adventure

Reply:
Being ultra picky, but this also won’t take into account stuff like the natural spawn protection minecraft has. There should be some method in the player class (like mayUseItemAt, may not be the right method but I believe there is a method that handles this) that should handle this logic for you nicely.

Reply:
Yeah ok